### PR TITLE
[Podman] Support for rootless containers for file log collection

### PR DESCRIPTION
**Pull request by tbavelier**
### What does this PR do?

* Permits the collection of log using file collection from podman containers run rootlessly (with a different user than `root`/ using `sudo`) by extracting the podman base path from the user-configured `podman_db_path` parameter

### Motivation

* https://dd.slack.com/archives/C046YU1V0SK/p1722853607862579
* Collection of logs from podman rootless containers using the host Agent

### Additional Notes

* As for rootfull containers, this requires enabling the podman socket and providing it to the Agent as `DOCKER_HOST` environment variable which is not really documented. A follow-up PR in `documentation` will be added.
* As stressed in our (future) documentation, this is not necessary when using the container Agent as the rootless podman base path can be mounted with a volume in the Agent to `/var/lib/containers` so it believes the container is run with the root user

### Describe how to test/QA your changes

QA'd myself with the steps below on top of the unit tests

1. Start a Ubuntu VM
2. Install podman with `sudo apt-get install podman -yq`
3. Enable the [root and user podman sockets](https://github.com/containers/podman/blob/main/docs/tutorials/socket_activation.md)
    ```bash
    systemctl --user start podman.socket
    sudo systemctl start podman.socket
    ```
4. **Rootfull container - no regression** : 
    - Add the root podman socket to the [Agent env variable](https://docs.datadoghq.com/agent/guide/environment-variables/#using-environment-variables-in-systemd-units) with `DOCKER_HOST=unix:///run/podman/podman.sock`
    - Grant Agent `rw` permissions to the socket with `sudo chmod o+rw /run/podman/podman.sock`.
        - `/run/podman` should already be available to `others` with `x` so `dd-agent` can enter the directory
    - Add in the Agent configuration `datadog.yaml` the `logs_config.use_podman_logs` parameter set to `true` (and ensure `logs_enabled` is also set to `true`)
    - Start a rootfull container logging `sudo podman run --log-driver=k8s-file --label com.datadoghq.ad.logs='[{"source": "bash_root", "service": "bash_root"}]' -d docker.io/library/bash:latest -c 'while true; do date; sleep 1; done'`
    - Grant permissions to the Linux `dd-agent` user with `sudo chmod -R o+rx /var/lib/containers/`
    - Restart the Agent to ensure it auto-discovers `podman` is running on the host and that it has the permissions to start the collector
    - Run `datadog-agent status` and ensure logs are picked up with the correct identifier, the container ID, the correct source and service from the labels
        image
    - Run `datadog-agent stream-logs` and ensure you get a message every second, e.g. the log from the container
        image
5. **Rootless container - enhancement** : 
    - Add the rootless podman socket to the [Agent env variable](https://docs.datadoghq.com/agent/guide/environment-variables/#using-environment-variables-in-systemd-units) with `DOCKER_HOST=unix:///run/podman/podman.sock`
        - The path depends on the user, so you can determine it with `podman info`
            ```bash
            ubuntu@ip-172-29-136-28:~$ podman info -f json | jq .host.remoteSocket.path
            "/run/user/1000/podman/podman.sock"
            ```
    - Grant Agent `rw` permissions to the socket with `sudo chmod o+rw /path/to/socket`.
        - Parent directories need to be be available to `others` with `x` so `dd-agent` can enter them
          ```bash
          - ubuntu@ip-172-29-136-28:~$ namei -m "/run/user/1000/podman/podman.sock"
          f: /run/user/1000/podman/podman.sock
           drwxr-xr-x /
           drwxr-xr-x run
           drwxr-xr-x user
           drwx---rwx 1000
           drwxr-xrwx podman
           srw-rw-rwx podman.sock
          ```
    - ~~Add in the Agent configuration `datadog.yaml` the `logs_config.use_podman_logs` parameter set to `true` (and ensure `logs_enabled` is also set to `true`)~~
        - This should already be present from testing rootfull
    - Start a rootless container logging `podman run --log-driver=k8s-file --label com.datadoghq.ad.logs='[{"source": "bash_rootless", "service": "bash_rootless"}]' -d docker.io/library/bash:latest -c 'while true; do date; sleep 1; done'`
    - Grant permissions to the Linux `dd-agent` user with `sudo chmod -R o+rx /path/to/database`
        - You can identify the base directory with `podman info -f json | jq .store.graphRoot`. If using SQLite as back-end (Podman 4.8+), there should be a `db.sql` file in it. If using less recent version, there is `libpod/bolt_state.db`
    - Configure in `datadog.yaml` the parameter `podman_db_path` with your podman db path from previous step
      ```bash
      ubuntu@ip-172-29-136-28:~/.local/share/containers/storage$ ls
      db.sql  defaultNetworkBackend  libpod  networks  overlay  overlay-containers  overlay-images  overlay-layers storage.lock  userns.lock
      ```
      So in my case `podman_db_path: "/home/ubuntu/.local/share/containers/storage/db.sql"`
    - Restart the Agent to ensure it auto-discovers `podman` is running on the host and that it has the permissions to start the collector when reviewing `podman_db_path`
    - Run `datadog-agent status` and ensure logs are picked up with the correct identifier, the container ID, the correct source and service from the labels
      image
    - Run `datadog-agent stream-logs` and ensure you get a message every second, e.g. the log from the container
      image

### CODE REVIEW

File: releasenotes/notes/podman_file_logs_path_rootless_support-55a1af8f41cf94c4.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+enhancements:
+  - |
+    Adds support for Podman rootless containers file log collection when ``logs_config.use_podman_logs`` is set ``true`` and ``podman_db_path`` is configured.
```
Comment (neko-dd) [inline releasenotes/notes/podman_file_logs_path_rootless_support-55a1af8f41cf94c4.yaml:4]:
```suggestion
    Adds support for file log collection from Podman rootless containers when ``logs_config.use_podman_logs`` is set to ``true`` and ``podman_db_path`` is configured.
```

**Review by carlosroman — APPROVED** [review]

File: pkg/util/log/log_podman_util.go
Comment (pgimalac) [inline pkg/util/log/log_podman_util.go:1]: Can you update codeowners for this file please ?

**Review by pgimalac — APPROVED** [review]

**Review by neko-dd — APPROVED** [review]

File: pkg/util/log/log_podman_util.go
Comment (tbavelier) [reply pkg/util/log/log_podman_util.go:1]: @pgimalac do you want this file to be co-owned by AML as `pkg/util/log` is under ASC ?

**Pull request merged**